crates/uv-git/src/git.rs

```diff
@@ -969,13 +969,34 @@ pub(crate) fn fetch(
 
     debug!("Performing a Git fetch for: {remote_url}");
     match strategy {
-        FetchStrategy::Cli => fetch_with_cli(repo, remote_url, &refspecs, tags),
+        FetchStrategy::Cli => {
+            // Try each refspec
+            let results = refspecs
+                .into_iter()
+                .map(|refspec| fetch_with_cli(repo, remote_url, &[refspec], tags))
+                .collect::<Vec<_>>();
```

Comment:
This could hide errors. It's unclear to me what happens inside libgit2 but it definitely succeeds with missing refspecs.

Alternatively, we could special case the `BranchOrTag` case and take the first matching refspec?

Reply:
Do we "typically" have multiple refspecs here? Like would we expect this to cause a noticeable slowdown, doing them by-one-one?

Reply:
It's just two cases afaik

- We have a short commit and fetch _everything_
- We have a branch or tag and fetch each of those (we only want the first one that is present though)